lib/nats.js

```diff
@@ -713,6 +713,11 @@ Client.prototype.setupHandlers = function () {
     return
   }
 
+  if (this.connectionTimeoutHandler) {
```

Comment:
To me, this looks like when the timeout is originally set (when connection is created), it will be reset when processing the INFO if we switch to TLS. Which means that the timeout will be possibly increased. I would simply have the check below include that the timer be set if this.connectionTimeoutHandler is "nil" (undefined).

Reply:
The timeout is cleared at that point, and a new one created. The old timeout will call destroy on the wrong socket. This is the correct syntax for the test.

Reply:
But my point was that when setupHandlers() is called a second time, `this.stream` has been recreated (for the tls client connection) and so we the timer fires (created for the original tcp connection), we still destroy this.stream, so that should apply to the current one. Otherwise, I don't understand Node and what's the point of "this.stream" if it references a closure that was set previously.. that would be very confusing to me.